From 08453710ea02a684c935330e55cc94fbabc58c54 Mon Sep 17 00:00:00 2001
From: Hariprasad Kelam <hkelam@marvell.com>
Date: Sun, 7 Mar 2021 17:45:30 +0530
Subject: [PATCH 15/17] Revert "octeontx2-pf: Add LIO3 link modes to ethtool's
 list of supported modes"

commit c7d93b72786f3782d6c9bb291ef1110b3a6deee9 from
git@git.assembla.com:cavium/WindRiver.linux.git

This reverts commit bd1bb56efdeb1b0a4fafa531fe083c485dd80c85.

Change-Id: I178f3a43afa224158a69038b2bf19977cb187844
Signed-off-by: Hariprasad Kelam <hkelam@marvell.com>
Reviewed-on: https://sj1git1.cavium.com/c/IP/SW/kernel/linux/+/47408
Reviewed-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Tested-by: Sunil Kovvuri Goutham <Sunil.Goutham@cavium.com>
Integrated-by: Abhishek Paliwal <paliwal.abhishek@windriver.com>
---
 drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c | 6 +++---
 1 file changed, 3 insertions(+), 3 deletions(-)

diff --git a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
index 89e645b..6e0cba1 100644
--- a/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
+++ b/drivers/net/ethernet/marvell/octeontx2/nic/otx2_ethtool.c
@@ -43,7 +43,7 @@ struct otx2_stat {
 }
 
 /* Physical link config */
-#define OTX2_ETHTOOL_SUPPORTED_MODES 0x638CFFF //110001110001100111111111111
+#define OTX2_ETHTOOL_SUPPORTED_MODES 0x630CDFF //110001100001100110111111111
 
 enum link_mode {
        OTX2_MODE_SUPPORTED,
@@ -1045,7 +1045,7 @@ static void otx2_get_link_mode_info(u64 link_mode_bmap,
 		ETHTOOL_LINK_MODE_20000baseMLD2_Full_BIT,
 		ETHTOOL_LINK_MODE_10000baseCR_Full_BIT,
 		ETHTOOL_LINK_MODE_25000baseSR_Full_BIT,
-		ETHTOOL_LINK_MODE_25000baseCR_Full_BIT,
+		ETHTOOL_LINK_MODE_20000baseKR2_Full_BIT,
 		ETHTOOL_LINK_MODE_25000baseCR_Full_BIT,
 		ETHTOOL_LINK_MODE_25000baseKR_Full_BIT,
 		ETHTOOL_LINK_MODE_40000baseSR4_Full_BIT,
@@ -1055,7 +1055,7 @@ static void otx2_get_link_mode_info(u64 link_mode_bmap,
 		ETHTOOL_LINK_MODE_10000baseSR_Full_BIT,
 		ETHTOOL_LINK_MODE_50000baseSR2_Full_BIT,
 		ETHTOOL_LINK_MODE_10000baseLR_Full_BIT,
-		ETHTOOL_LINK_MODE_50000baseCR2_Full_BIT,
+		ETHTOOL_LINK_MODE_56000baseKR4_Full_BIT,
 		ETHTOOL_LINK_MODE_50000baseCR2_Full_BIT,
 		ETHTOOL_LINK_MODE_50000baseKR2_Full_BIT,
 		ETHTOOL_LINK_MODE_10000baseLRM_Full_BIT,
-- 
1.9.1

